### PR TITLE
[Strings] Remove operations not included in imported strings

### PR DESCRIPTION
**Pull request by tlively**
The stringref proposal has been superseded by the imported JS strings proposal,
but the former has many more operations than the latter. To reduce complexity,
remove all operations that are part of stringref but not part of imported
strings.

### CODE REVIEW

**Comment by tlively** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6589?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6589** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6589?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6579** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6579?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6583** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6583?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tlively and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by kripken** [comment]
I would personally be more cautious about removing stuff, since imported strings may expand :smile: but sgtm if you are willing to restore stuff that ends up necessary.

**Comment by tlively** [comment]
Yep, I can sign myself up for that.

File: test/lit/passes/simplify-locals-strings.wast [hunk 1]
```diff
@@ -10,185 +10,9 @@
...
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (string.new_utf8
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT: )
-  (func $yes-new-past-store
```
Comment (kripken) [inline test/lit/passes/simplify-locals-strings.wast:140]: We don't seem to have any `yes-new-*` functions after this removal, so we might be losing coverage here.

**Review by kripken — COMMENTED** [review]
lgtm % comment

**Comment by tlively** [comment]
### Merge activity

* **May 15, 3:05 PM EDT**: @tlively started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6589).

**Pull request merged**